### DISCUSSION TITLE
How to export xrepo packages to PATH and LD_LIBRARY_PATH ?

### DISCUSSION THREAD

**Discussion by yh-sb** [xmake-io/xmake#5494] [Q&A] [answered]
I am using xmake project with xrepo packages: qt6core and qt6gui. Build works fine.
Then I was adding linuxdeploy to get .AppImage file.

linuxdeploy tool need to have paths to search qmake, libQt6Gui.so and so on.

Currently I am doing it like this:
```bash
QT6BASE=$(xrepo info qt6base | grep 'installdir.*qt6base' | awk '{print $4}')
export PATH=$QT6BASE/bin:$PATH
export LD_LIBRARY_PATH=$QT6BASE/lib:$LD_LIBRARY_PATH
linuxdeploy.AppImage --appdir appdir --desktop-file ../app.desktop --icon-file ../icon.png --plugin qt --output appimage
```

But it's very inconvenient and unreliable!
**Is there any elegant way to environment with all path set to run external command to deploy?** Like `xmake run`, but for deployment.

**Comment by waruqi** [comment, marked as answer]
use `xrepo env` and `xrepo env shell`
